code/parse/sexp.cpp

```diff
@@ -1968,6 +1963,33 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 			continue; 
 		}
 
+		// if this is the special argument, make sure it is being properly used
+		if (type2 == SEXP_ATOM_STRING && Sexp_nodes[node].flags & SNF_SPECIAL_ARG_IN_NODE) {
+			bool found = false;
+			z = node;
+			while (true) {
+				z = find_parent_operator(z);
+				if (z < 0) {
+					break;
+				}
+				if (is_blank_argument_op(get_operator_const(z))) {
+					found = true;
+					break;
+				}
+			}
+
+			// if there is no argument operator higher in the tree, we have a problem
+			if (!found) {
```

Comment:
Optional: I think starting with the positive case is easier to read. I was initially confused about the code, because I missed the `!` and thought that finding a blank operator was the negative case. It's not important, though, and partly a matter of personal taste.

Reply:
I originally had the if() block the other way around, and then I modified it to what you see here.  The comments got a little messy the other way.